tests/unit/connect-test.js

```diff
@@ -59,3 +59,35 @@ test('connect with remote being promise', function (t) {
 
   t.end()
 })
+
+test('connect replication error', function (t) {
+  t.plan(3)
+
+  var syncApi = {
+    on: simple.stub()
+  }
+  var state = {
+    emitter: {
+      emit: simple.stub()
+    },
+    remote: 'remoteDb',
+    db: {
+      sync: simple.stub()
+    }
+  }
+  simple.mock(state.db, 'sync').returnWith(syncApi)
+
+  connect(state)
+
+  .then(function () {
+    t.is(syncApi.on.calls[0].args[0], 'error', 'listens "error" event')
```

Comment:
But in this case `state.replication.on` is called twice -- first to bind the 'error' event and then to bind the 'change' event. Since we want to access the 'error' event `on` call, we need to access the first call instead of the last call, which I why I used `.calls[0].args[0]`.  Or is that not right?